docs/source/_static/data_processing_examples/starmap.py

```diff
@@ -0,0 +1,192 @@
+#
+# To examine this data, the vignette displays the max projection of channels and
+# rounds. Ideally, these should form fairly coherent spots, indicating that the
+# data are well registered. By contrast, if there are patterns whereby pairs of
+# spots are consistently present at small shifts, that can indicate systematic
+# registration offsets which should be corrected prior to analysis.
+
+ch_r_projection = stack.max_proj(Axes.CH, Axes.ROUND)
+
+f, ax = plt.subplots()
+imshow_plane(ch_r_projection, sel={Axes.ZPLANE: 15}, ax=ax)
+
+###############################################################################
+# Visualize the codebook
+# ----------------------
+# The STARmap codebook maps pixel intensities across the rounds and channels to
+# the corresponding barcodes and genes that those pixels code for. For this
+# dataset, the codebook specifies 160 gene targets.
+
+print(experiment.codebook)
+
+###############################################################################
+# Registration
+# ------------
+# Starfish exposes some simple tooling to identify registration shifts.
+# :code:`starfish.util.plot.diagnose_registration` takes an ImageStack and a
+# set of selectors, each of which maps :py:class:`~starfish.types.Axes` objects
+# to indices that specify a particular 2d image.
+#
+# Below the vignette projects the channels and z-planes and examines the
+# registration of those max projections across channels 0 and 1. To make the
+# difference more obvious, we zoom in by selecting a subset of the image, and
+# display the data before and after registration.
+#
+# It looks like there is a small shift approximately the size of a spot
+# in the `x = -y` direction for at least the plotted rounds
+#
+# The starfish package can attempt a translation registration to fix this
+# registration error.
+#
+# Register the data:
+projection = stack.max_proj(Axes.CH, Axes.ZPLANE)
+reference_image = projection.sel({Axes.ROUND: 1})
+
+ltt = starfish.image.LearnTransform.Translation(
+    reference_stack=reference_image,
+    axes=Axes.ROUND,
+    upsampling=1000,
+)
+transforms = ltt.run(projection)
+
+###############################################################################
+# Apply and print out the size and direction of the transforms.
+
+print([t[2].translation for t in transforms.transforms])
+
+warp = starfish.image.ApplyTransform.Warp()
+stack = warp.run(
+    stack=stack,
+    transforms_list=transforms,
+)
+
+###############################################################################
+# Show the effect of registration.
+
+post_projection = stack.max_proj(Axes.CH, Axes.ZPLANE)
+
+f, (ax1, ax2) = plt.subplots(ncols=2)
+sel_0 = {Axes.ROUND: 0, Axes.X: (500, 600), Axes.Y: (500, 600)}
+sel_1 = {Axes.ROUND: 1, Axes.X: (500, 600), Axes.Y: (500, 600)}
+diagnose_registration(
+    projection, sel_0, sel_1, ax=ax1, title='pre-registered'
+)
+diagnose_registration(
+    post_projection, sel_0, sel_1, ax=ax2, title='registered'
```

Comment:
in the checked in notebook, this title still says 'pre-registered' -- looks like the .ipynb is out of sync w/ the py representation. i'm guessing this has to do with two sources of truth: sphinx gallery, and ipynb?